src/main/java/org/jabref/model/database/BibDatabaseContext.java

```diff
@@ -286,4 +293,13 @@ public boolean equals(Object o) {
     public int hashCode() {
         return Objects.hash(database, metaData, path, location);
     }
+
+    /**
+     * Get the generated UID for the current context. Can be used to distinguish contexts with changing metadata etc
+     *
+     * @return The generated UID in UUIDv4 format
+     */
+    public String getUid() {
+        return uid;
```

Comment:
The usage of a plain UUID as a key was confusing while using the debugger, it wasn't intuitive to know what database/library it was referring to:

![image](https://github.com/JabRef/jabref/assets/21198231/87787079-7774-46c5-9250-a51e07810bd7)


Maybe we do something like `{databasename}_{uid}`?

Reply:
I was thinking about such things as well and tried some things, however 
1. A new unsaved database doesn't have a path
2. If it's a shared database it doesn't have a path as well, it has a shared database
3. Saving under a new path/name will change the key, and we lose the reference

However, I now added the prefix bibdatabasecontext to. make it a bit more clear where the uid is related to

Reply:
How about? 
```python
if databasename is empty:
     uid = "bibdatabasecontext_${uid}"
else 
    uid = "${databasename}_${uid}"
```

Reply:
> Saving under a new path/name will change the key, and we lose the reference

Ah okay I see, I'm fine with the `bibdatabasecontext_` prefix for now but we really should consider other options later.